cmake/Platform/Mac/InstallUtils_mac.cmake.in

```diff
@@ -0,0 +1,146 @@
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+function(fixup_qt_framework lib_name framework_path)
+
+    file(REMOVE_RECURSE
+        ${framework_path}/Headers
+        ${framework_path}/Resources
+        ${framework_path}/${lib_name}
+        ${framework_path}/Versions/Current
+        ${framework_path}/Versions/5/Headers
+    )
+
+    execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink 5 Current
+        WORKING_DIRECTORY ${framework_path}/Versions
+    )
+    execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink Versions/5/${lib_name} ${lib_name}
+        WORKING_DIRECTORY ${framework_path}
+    )
+
+endfunction()
+
+function(fixup_python_framework framework_path)
+
+    file(REMOVE_RECURSE
+        ${framework_path}/Versions/Current
+        ${framework_path}/Versions/3.7/Headers
+        ${framework_path}/Versions/3.7/lib/Python
+        ${framework_path}/Versions/3.7/lib/python3.7/test
+        ${framework_path}/Versions/3.7/lib/python3.7/site-packages/scipy/io/tests
+        ${framework_path}/Python
+        ${framework_path}/Resources
+        ${framework_path}/Headers
+    )
+    
+    file(GLOB_RECURSE exe_file_list "${framework_path}/**/*.exe")
+    if(exe_file_list)
+        file(REMOVE_RECURSE ${exe_file_list})
+    endif()
+    execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink include/python3.7m Headers
+        WORKING_DIRECTORY ${framework_path}/Versions/3.7
+    )
+    execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink 3.7 Current
+        WORKING_DIRECTORY ${framework_path}/Versions/
+    )
+    execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink Versions/Current/Python Python
+        WORKING_DIRECTORY ${framework_path}
+    )
+    execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink Versions/Current/Headers Headers
+        WORKING_DIRECTORY ${framework_path}
+    )
+    execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink Versions/Current/Resources Resources
+        WORKING_DIRECTORY ${framework_path}
+    )
+    file(CHMOD ${framework_path}/Versions/Current/Python 
+        PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_WRITE GROUP_EXECUTE WORLD_READ WORLD_EXECUTE 
+    )
+
+endfunction()
+
+function(codesign_file file entitlement_file)
+
+    if(EXISTS ${entitlement_file})
+
+        execute_process(COMMAND "/usr/bin/codesign" "--force" "--sign" "@LY_CODE_SIGN_IDENTITY@" "--deep" "-o" "runtime" "--timestamp" "--entitlements" "${entitlement_file}" "${file}"
+            TIMEOUT 300
+            OUTPUT_VARIABLE codesign_out
+            RESULT_VARIABLE codesign_ret
+        )
+    else()
+        execute_process(COMMAND "/usr/bin/codesign" "--force" "--sign" "@LY_CODE_SIGN_IDENTITY@" "--deep" "-o" "runtime" "--timestamp" "${file}"
+            TIMEOUT 300
+            OUTPUT_VARIABLE codesign_out
+            RESULT_VARIABLE codesign_ret
+        )
+    endif()
```

Comment:
Maybe consolidate to something like:
```cmake
set(code_sign_command "/usr/bin/codesign" "--force" "--sign" "@LY_CODE_SIGN_IDENTITY@" "--deep" "-o" "runtime" "--timestamp")

if(EXISTS ${entitlement_file})
    list(APPEND code_sign_command "--entitlements" "${entitlement_file}")
endif()

execute_process(COMMAND ${code_sign_command} "${file}"
    TIMEOUT 300
    OUTPUT_VARIABLE codesign_out
    RESULT_VARIABLE codesign_ret
)
```

Reply:
I tried that, but it doesn't work. The execute_process command doesn't  pass in the parameters correctly if it's part of the same string. It gives errors like:
`codesign: unrecognized option `--force;--sign;<signing_identity>;-o;runtime;--timestamp;--entitlements;/Users/srikkant/O3DE/o3de/build/mac/Code/Tools/AssetProcessor/O3DE.build/profile/AssetProcessor.build/AssetProcessor.app.xcent'` 

or if I change the ";" to spaces:

`codesign: unrecognized option `--force --sign <signing_identity> -o runtime --timestamp --entitlements /Users/srikkant/O3DE/o3de/build/mac/Code/Tools/AssetProcessor/O3DE.build/profile/AssetProcessor.build/AssetProcessor.app.xcent'`

Reply:
Remove the double quotes from the execute_process command "${file}"